de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend

```diff
@@ -48,16 +51,31 @@ class OvertargetQuickfixProvider extends DefaultQuickfixProvider {
 	
 	@Fix(Diagnostic.LINKING_DIAGNOSTIC)
 	def fixCannotResolveReference(Issue issue, IssueResolutionAcceptor acceptor) {
-		acceptor.accept(issue, 'Generate Target for References', '', '', 
+		acceptor.accept(issue, 'Generate targetForReferences to resolve references', 'Generates a targetForReferences and sets it as active target. \n After that the actual target, which could not be resolved, is set as active target.', '', 
 			new IModification() {
 			
 			override apply(IModificationContext context) throws Exception {
 				if (issue.message.contains("Couldn't resolve reference to")) {
+					val genHandler = new GenerationHandler();
+					val refTargetHelper = new ReferencedTargetHelper()
+					val targetPlatHelper = new TargetPlatformHelper()
+				
 					val editor = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor()
 					if (editor instanceof ITextEditor) {
 						val ite = editor as ITextEditor
 						val input = ite.editorInput
-						new GenerationHandler().runGeneration(input);
+						genHandler.runGeneration(input);
+						
+						//find targetForReferences.target in directory and set it as active target
+						val fileEditorInput = input as IFileEditorInput
+						val file = fileEditorInput.file
+						val outputDirectory = genHandler.getOutputConfigurations(input)
+						val targetForReferencesFile = refTargetHelper.findTargetForReferencesFile(file, outputDirectory)
+						targetPlatHelper.asActiveTarget = targetForReferencesFile;
```

Comment:
Is it also resolving the target here?

Reply:
not here, but in the TargetPlatformHelper